examples/src/main/java/com/mapbox/navigation/examples/core/SilentWaypointsRerouteActivity.kt

```diff
@@ -0,0 +1,338 @@
+            getRerouteController()?.unregisterRerouteStateObserver(
+                this@SilentWaypointsRerouteActivity
+            )
+        }
+        mapView.onStop()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapboxReplayer.finish()
+        mapboxNavigation?.stopTripSession()
+        mapboxNavigation?.onDestroy()
+        mapView.onDestroy()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        mapView.onLowMemory()
+    }
+
+    @SuppressLint("MissingPermission")
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { _ ->
+            mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(15.0))
+            navigationMapboxMap = NavigationMapboxMap.Builder(mapView, mapboxMap, this)
+                .vanishRouteLineEnabled(true)
+                .build()
+
+            // Center the map at current location. Using LocationEngineProvider because the
+            // replay engine won't have your last location.
+            LocationEngineProvider
+                .getBestLocationEngine(this)
+                .getLastLocation(locationListenerCallback)
+
+            directionRoute?.let {
+                navigationMapboxMap?.drawRoute(it)
+                mapboxNavigation?.setRoutes(listOf(it))
+                btnStartNavigation.visibility = VISIBLE
+            }
+        }
+
+        mapboxMap.addOnMapLongClickListener { latLng ->
+            waypointsController.add(latLng)
+            mapboxMap.locationComponent.lastKnownLocation?.let { originLocation ->
+                mapboxNavigation?.requestRoutes(
+                    RouteOptions.builder().applyDefaultParams()
+                        .accessToken(Utils.getMapboxAccessToken(applicationContext))
+                        .coordinates(waypointsController.coordinates(originLocation))
+                        .waypointIndices("0;${waypointsController.waypoints.size}")
+                        .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                        .build(),
+                    routesReqCallback
+                )
+            }
+            true
+        }
+
+        if (directionRoute == null) {
+            Snackbar.make(container, R.string.msg_long_press_map_to_place_waypoint, LENGTH_SHORT)
+                .show()
+        }
+    }
+
+    @SuppressLint("MissingPermission")
+    private fun initListeners() {
+        setupReplayControls()
+        btnStartNavigation.setOnClickListener {
+            updateCameraOnNavigationStateChange(true)
+            navigationMapboxMap?.addProgressChangeListener(mapboxNavigation!!)
+            if (mapboxNavigation?.getRoutes()?.isNotEmpty() == true) {
+                navigationMapboxMap?.startCamera(mapboxNavigation?.getRoutes()!![0])
+            }
+            mapboxNavigation?.registerRouteProgressObserver(ReplayProgressObserver(mapboxReplayer))
```

Comment:
could we make it depends on external setting `simulate route` as at `SimpleMapboxNavigationKt`, to have options ride or simulate route
https://github.com/mapbox/mapbox-navigation-android/blob/7f56d62164c226b1da499f99972a0e0c5c168030/examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt#L217-L221